### PR TITLE
JBMAR-190 fixes for branch 1.4

### PR DESCRIPTION
**Pull request by ropalka**
Bringing in Multi-Release jar support for JBoss Marshalling + fixing OSGi build issue

### CODE REVIEW

**Comment by slaskawi** [comment]
Looks good regarding to the OSGi headers. Thanks a lot @ropalka!

**Comment by slaskawi** [comment]
Hey @ropalka ! 

May I ask you to cherry-pick this fix into 2.0 branch and perform a release? We [have several tests blocked for a very long time waiting for OSGi hearers](https://issues.jboss.org/browse/JBMAR-188).

Thanks!
Sebastian

**Comment by dmlloyd** [comment]
Conflicts

File: api/src/main/java/org/jboss/marshalling/reflect/JDKSpecific.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+final class JDKSpecific {
+
+    private static final ReflectionFactory reflectionFactory = AccessController.doPrivileged(new PrivilegedAction<ReflectionFactory>() {
+        public ReflectionFactory run() { return ReflectionFactory.getReflectionFactory(); }
+    });
+
+    static Constructor<?> newConstructorForSerialization(Class<?> classToInstantiate, Constructor<?> constructorToCall) {
+        return reflectionFactory.newConstructorForSerialization(classToInstantiate, constructorToCall);
```
Comment (jglick) [inline api/src/main/java/org/jboss/marshalling/reflect/JDKSpecific.java]: Apparently overridden with [this](https://github.com/jboss-remoting/jboss-marshalling-jdk9/blob/c4f02d58969c742dd2a177c7b22545fac48f16d1/src/main/java/org/jboss/marshalling/reflect/JDKSpecific.java#L38). Would it not have been a lot simpler to just use Java reflection to call the JDK 9 version where available?

Comment (dmlloyd) [inline api/src/main/java/org/jboss/marshalling/reflect/JDKSpecific.java]: There are many more changes which will be required for full Java 9 module support, so we might as well do it the "right" way now.

**Pull request merged**